agent/app/agent_capability.go

```diff
@@ -316,6 +342,50 @@ func (agent *ecsAgent) appendTaskENICapabilities(capabilities []*ecs.Attribute)
 	return capabilities
 }
 
+func (agent *ecsAgent) appendExecCapabilities(capabilities []*ecs.Attribute) ([]*ecs.Attribute, error) {
```

Comment:
I think we should probably test that config directory is writable, and maybe also that the binaries and certs are read only?

Reply:
Also, this is something we haven't discussed thoroughly, so maybe it's for a new PR, but calling it out here... I think we should check that the binaries we're taking are legit (exec signature or md5 or somethig)... these are binaries we're mounting into customer's containers, so we want to make sure we won't mount some spurious binary that could wreak havoc in the customer's container.

Reply:
good point, agree that it'd be better for us to add that functionality (validating the binaries) as an add-on after discussion

Reply:
for checking if `/config` is writable while `/bin` and `/certs` are read-only, golang does not seem to provide a native way for this check. 1 possible way (still testing and prototyping) is to use the unix permission bits, but I'm not sure if that would work for Windows scenarios or if that (Windows scenario) is something we should care about

Reply:
Wonder if something like this works:
https://stackoverflow.com/questions/45429210/how-do-i-check-a-files-permissions-in-linux-using-go
or
https://github.com/phayes/permbits/blob/master/permbits.go

Reply:
did some prototyping and here are what I found: regardless of what the bind mount is specified as `readonly` or not, the permission bits are the same (with `ls -l`). here are the steps I took:

start container with bind mounts:
```
$ docker run -it \
--mount type=bind,src="${HOME}/ro_root",dst=/exec_test/ro_root,readonly \
--mount type=bind,src="${HOME}/rw_root",dst=/exec_test/rw_root \
--mount type=bind,src="${HOME}/ro_file",dst=/exec_test/test_dir/ro_file,readonly \
--mount type=bind,src="${HOME}/rw_file",dst=/exec_test/test_dir/rw_file \
amazonlinux:latest
```

results:
```
$ ls -lR /exec_test
/exec_test:
total 0
drwxrwxr-x 3 1000 1000 35 Oct 28 22:03 ro_root
drwxrwxr-x 3 1000 1000 35 Oct 28 22:04 rw_root
drwxr-xr-x 2 root root 36 Oct 28 22:09 test_dir

/exec_test/ro_root:
total 0
drwxrwxr-x 2 1000 1000 18 Oct 28 22:03 child_dir
-rw-rw-r-- 1 1000 1000  0 Oct 28 22:03 file

/exec_test/ro_root/child_dir:
total 0
-rw-rw-r-- 1 1000 1000 0 Oct 28 22:03 file

/exec_test/rw_root:
total 0
drwxrwxr-x 2 1000 1000 18 Oct 28 22:04 child_dir
-rw-rw-r-- 1 1000 1000  0 Oct 28 22:04 file

/exec_test/rw_root/child_dir:
total 0
-rw-rw-r-- 1 1000 1000 0 Oct 28 22:04 file

/exec_test/test_dir:
total 0
-rw-rw-r-- 1 1000 1000 0 Oct 28 22:04 ro_file
-rw-rw-r-- 1 1000 1000 0 Oct 28 22:06 rw_file
```

as indicated here, both files and directories (readonly and read-write) have the same permission bits (`-rw`, `drw`), don't think we can check this only with permission bits

Reply:
the error happens at runtime, e.g.

```
$ echo "hello world" > /exec_test/test_dir/ro_file
bash: ro_file: Read-only file system
```

Reply:
I think we can only check for existence (if the binaries and cert file exist) in this change, I will look further into how to check if a bind-mounted file/directory is read-only and update the check logic with a follow-up PR if it's feasible

Reply:
ok, how about just attempting to write a temp file into this directory? which should succeed...

we can do this n a new PR as you suggest

Reply:
sounds good! thanks!